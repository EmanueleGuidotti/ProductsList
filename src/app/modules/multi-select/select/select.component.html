

        <h4>{{title}}</h4>
        <mat-form-field>
            <mat-select (selectionChange)="sendValue($event)">
                <mat-option *ngFor="let vehicle of vehicles" [value]="vehicle.type">
                    {{vehicle.type}}
                </mat-option>
            </mat-select>
        </mat-form-field>


