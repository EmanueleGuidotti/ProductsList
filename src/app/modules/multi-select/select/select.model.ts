export class SelectModel {
    type: string;
}
