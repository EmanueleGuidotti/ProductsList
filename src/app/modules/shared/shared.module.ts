import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatSelectModule, MatGridListModule } from '@angular/material';
import { MatListModule } from '@angular/material/list';
import {MatCardModule} from "@angular/material/card";

@NgModule({
	declarations: [],
	imports: [CommonModule, MatSelectModule, MatGridListModule, MatListModule, MatCardModule],
	exports: [MatSelectModule, MatGridListModule, MatListModule, MatCardModule],
	providers: []
})
export class SharedModule {}
