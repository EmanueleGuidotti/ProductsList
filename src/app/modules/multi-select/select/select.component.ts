import {
    Component,
    ChangeDetectionStrategy,
    Input,
    ChangeDetectorRef, EventEmitter, Output,
} from '@angular/core';
import { SelectModel } from './select.model';
import {MatSelectChange} from "@angular/material/select";

@Component({
	selector: 'select-component',
	templateUrl: './select.component.html',
	styleUrls: ['./select.component.scss'],
	changeDetection: ChangeDetectionStrategy.OnPush
})
export class SelectComponent {
	@Input() vehicles: SelectModel[];
	@Input() title: string;
    n:string="";

	constructor(public cdf: ChangeDetectorRef) {}

	public sendValue(n: MatSelectChange) {
        this.n = n.source.value;
    }
}
