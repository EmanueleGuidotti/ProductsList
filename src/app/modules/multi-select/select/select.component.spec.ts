import { async, ComponentFixture, flush, TestBed } from '@angular/core/testing';
import { SelectComponent } from './select.component';
import { SharedModule } from '../../shared/shared.module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { SelectModel } from './select.model';
import { By } from '@angular/platform-browser';
import { DebugElement } from '@angular/core';

describe('SelectComponent', () => {
	let component: SelectComponent;
	let fixture: ComponentFixture<SelectComponent>;
	let debugElement: DebugElement;

	const vehicles: SelectModel[] = [
		{ type: 'car' },
		{ type: 'airplane' },
		{ type: 'train' }
	];

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [SharedModule, BrowserAnimationsModule],
			declarations: [SelectComponent]
		}).compileComponents();
		fixture = TestBed.createComponent(SelectComponent);
		component = fixture.componentInstance;
		debugElement = fixture.debugElement;
	}));

	it('should create the component', () => {
		const selectInstance = debugElement.componentInstance;
		component.vehicles = vehicles;
		component.title = 'Vehicles Types';
		component.cdf.markForCheck();
		fixture.autoDetectChanges(true);
		expect(component).toBeTruthy();
		expect(selectInstance).toBeTruthy();
	});

	it(`should select render 3 options`, async () => {
		component.vehicles = vehicles;
		component.title = 'Vehicles Types';
		component.cdf.markForCheck();
		fixture.autoDetectChanges(true);
		const trigger = debugElement.query(By.css('.mat-select-trigger'));
		trigger.triggerEventHandler('click', null);
		await fixture.whenStable().then(() => {
			const inquiryOptions = debugElement.queryAll(By.css('.mat-option-text'));
			expect(inquiryOptions.length).toEqual(3);
		});
	});

	it(`should the first option is Car`, async () => {
		component.vehicles = vehicles;
		component.title = 'Vehicles Types';
		component.cdf.markForCheck();
		fixture.autoDetectChanges(true);
		const trigger = debugElement.query(By.css('.mat-select-trigger'));
		trigger.triggerEventHandler('click', null);
		await fixture.whenStable().then(async () => {
			const options = debugElement.queryAll(By.css('.mat-option'));
			options[0].triggerEventHandler('click', null);
			const element = debugElement.query(By.css('.mat-select-value-text span'));
			expect(element.nativeElement.textContent).toContain('car');
			expect(fixture).toMatchSnapshot();
		});
	});
});
