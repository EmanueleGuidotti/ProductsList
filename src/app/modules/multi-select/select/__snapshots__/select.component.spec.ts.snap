// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent should the first option is Car 1`] = `
<select-component
  cdf={[Function ViewRef_]}
  n={[Function String]}
  title={[Function String]}
  vehicles={[Function Array]}
>
  <h4>
    Vehicles Types
  </h4><mat-form-field
    class="mat-form-field ng-tns-c8-0 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder mat-form-field-should-float mat-focused"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <mat-select
            aria-disabled="false"
            aria-invalid="false"
            aria-multiselectable="false"
            aria-required="false"
            class="mat-select ng-tns-c9-1 ng-star-inserted"
            id="mat-select-2"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-hidden="true"
              cdk-overlay-origin=""
              class="mat-select-trigger"
            >
              <div
                class="mat-select-value"
              >
                
                
                <span
                  class="mat-select-value-text ng-tns-c9-1 ng-star-inserted"
                >
                  
                  <span
                    class="ng-tns-c9-1 ng-star-inserted"
                  >
                    car
                  </span>
                  
                </span>
              </div>
              <div
                class="mat-select-arrow-wrapper"
              >
                <div
                  class="mat-select-arrow"
                />
              </div>
            </div>
            
          </mat-select>
          <span
            class="mat-form-field-label-wrapper"
          >
            
          </span>
        </div>
        
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c8-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c8-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field>
</select-component>
`;
